frontend/src/components/Table/Table.tsx

```diff
@@ -204,35 +202,24 @@ const IndeterminateCheckbox = React.forwardRef(
       rest.onChange && !allDataRef && rest.onChange(e);
       const currentSelected = selectedRef?.current ? [...selectedRef?.current] : [];
       if (isHeaderCheck) {
-        setSelected([]);
-      } else {
-        if (isSingleSelect === true) {
-          currentSelected.splice(0, currentSelected.length);
-          currentSelected.push(row.original);
+        if (e.target.checked && !indeterminate) {
```

Comment:
I was able to greatly simplify this logic. Need to add unit tests here as this logic has been unstable in the past.